integration-tests/jpa/src/main/java/io/quarkus/it/jpa/configurationless/CRUDResource.java

```diff
@@ -76,10 +79,10 @@
     @Transactional
     @Path("/import")
     public Map<String, String> get() {
+        boolean isProdMode = ProfileManager.getActiveProfile().equals(LaunchMode.NORMAL.getDefaultProfile());
```

Comment:
I'm not very satisfied with this but I didn't find a cleaner way to fix this test...

Reply:
Is it because the native tests are running under the `prod` profile?

Reply:
Now that you asked the question @machi1990, this might be linked to the #3154 reversal in #3265... (even if I tested #3265 in native mode and everything seemed fine at that time)... This needs to be investigated.

Reply:
I'm pretty sure reverting #3154 was a bad idea so I put it back (a modified version since there's now the `quarkus.test.profile` property to consider) and I got positive results locally on tests done from a quickstarts project, but CI is still not happy (the native test is running under the `prod` profile)...

Sorry about reverting your commit @machi1990 :)

Reply:
> I'm pretty sure reverting #3154 was a bad idea so I put it back (a modified version since there's now the `quarkus.test.profile` property to consider) and I got positive results locally on tests done from a quickstarts project, but CI is still not happy (the native test is running under the `prod` profile)...

I think this is because the image is generated with a `prod` profile. Reverting the commit in question will only help with properties that are available at runtime. 

I think we'll need investigating a bit more: @stuartwdouglas detailed the problem better here https://github.com/quarkusio/quarkus/pull/3154#issuecomment-509878882.

Reply:
@gsmet I have a few questions for you. I've been dealing with a native tests profile issue here which will require further investigations.

- I think we should limit this PR content to the original issue only and accept the `CRUDResource` workaround. WDYT?
- With the current `master` code, native tests are always run under the `prod` profile because I reverted #3154 in #3265. If we put back #3154 (the current version of `NativeImageLauncher` in this PR), we will have limited support for `test` profile in native tests, depending on when the profile is used (build time or runtime). Should we put it back or wait for the investigations and offer a full `test` profile support only if it's possible and nothing otherwise? In both cases, I think we need to put a warning in the `maven-tooling` and `gradle-tooling` guides about this native tests issue.

Reply:
Here's the `CRUDResource` workaround since it's not clear in the current PR content:
```
    public Map<String, String> get() {
        boolean isProdMode = ProfileManager.getActiveProfile().equals(LaunchMode.NORMAL.getDefaultProfile());
        Gift gift = em.find(Gift.class, 100000L);
        Map<String, String> map = new HashMap<>();
        map.put("jpa", gift != null || isProdMode ? "OK" : "Boooo");
        return map;
    }
```
I had to add `isProdMode` to make the native test pass.